src/app/core/views/Preferences/tabs/Plans/PlanSelector.tsx

```diff
@@ -206,3 +226,130 @@ function Price({
     </div>
   );
 }
+
+const ChangePlanDialog = ({
+  prices,
+  isDialgOpen,
+  setIsDialogOpen,
+  onPlanClick,
+  priceIdSelected,
+}: {
+  prices;
+  isDialgOpen: boolean;
+  setIsDialogOpen: (value: boolean) => void;
+  onPlanClick: (value: string) => void;
+  priceIdSelected: string;
+}): JSX.Element => {
+  const plan = useSelector<RootState, PlanState>((state) => state.plan);
+  const { translate } = useTranslationContext();
+
+  const { planLimit, planUsage, subscription } = plan;
+  const selectedPlan = prices.filter((plan) => plan.id === priceIdSelected);
```

Comment:
If the selected plan can only be used once it is better to use find, there is no need for an array to store a single value.

Reply:
done